backend/hdf5/h5x/H5DataType.cpp

```diff
@@ -281,6 +281,12 @@ data_type_from_h5(H5T_class_t vclass, size_t vsize, H5T_sign_t vsign)
     return DataType::Nothing;
 }
 
+void check_bool_enum(h5x::DataType type) {
+    assert(type.member_count() == 2);
```

Comment:
On a second though, you should be able to use `boolfiletype.enum_equal(type)` for the check.